### DISCUSSION TITLE
Dynamic add/remove rows in table with Tom-Select

### DISCUSSION THREAD

**Discussion by stuartfarquhar** [livewire/livewire#4310] [1. Help]
I am having trouble dynamically adding rows to a table with Tom Select. When adding a new row the Tom Select doesn't load and reverts to the default HTML Select field. 

Before:
<img width="1191" alt="Screen Shot 2021-11-30 at 9 30 12 AM" src="https://user-images.githubusercontent.com/86656810/143938491-23abc72c-4165-49c7-a1cc-aef9f0e455a6.png">

After:
<img width="1192" alt="Screen Shot 2021-11-30 at 9 30 23 AM" src="https://user-images.githubusercontent.com/86656810/143938512-bc637e9f-e8f7-4689-964f-a92d9923c95e.png">

Livewire Template code:

`<div>
   
    <div class="mt-4 box p-4 w-full h-96">

        <div class="flex justify-between">
            <div>Add New Purchase Item(s)</div>
            <button wire:click="addRow" class="btn btn-primary">
                + Add Row
            </button>
        </div>

        <div class="overflow-x-auto overflow-y-auto min-h-screen">
            <table class="table">
                <thead>
                <tr>
                    <th class="border-b-2 whitespace-nowrap">Name</th>
                    <th class="border-b-2 whitespace-nowrap">Material / Product</th>
                    <th class="border-b-2 whitespace-nowrap">Location</th>
                    <th class="border-b-2 whitespace-nowrap">Action</th>
                </tr>
                </thead>
                <tbody>
                @foreach ($rows as $index => $row)
                    <tr wire:key="index-{{ $index }}">
                        <td class="whitespace-nowrap">
                            <x-text-input wire:model="rows.{{$index}}.name" class="form-control"/>
                            <x-jet-input-error for="rows.{{$index}}.name" class="text-theme-6"/>
                        </td>
                        <td class="whitespace-nowrap">
                            <div wire:ignore>
                                <select id="rows.{{$index}}.stock"
                                        placeholder="Please select"
                                        class="tom-select w-full"
                                        wire:model="rows.{{$index}}.stock">
                                    <optgroup label="Products">
                                        @foreach ($all_products as $product)
                                            <option wire:key="p-{{ $product->id }}" value="{{ $product->id }}">
                                                {{ $product->name }}
                                            </option>
                                        @endforeach
                                    </optgroup>
                                    <optgroup label="Materials">
                                        @foreach ($all_materials as $material)
                                            <option wire:key="m-{{ $material->id }}" value="{{ $material->id }}">
                                                {{ $material->name }}
                                            </option>
                                        @endforeach
                                    </optgroup>
                                </select>
                            </div>
                        </td>
                        <td class="whitespace-nowrap">
                            <div wire:ignore>
                                <select id="rows.{{$index}}.location"
                                        placeholder="Please Select"
                                        class="tom-select w-full"
                                        wire:model="rows.{{$index}}.location">
                                    <option value=""></option>
                                    @foreach ($all_locations as $location)
                                        <option wire:key="l-{{ $location->id }}"
                                                value="{{ $location->id }}"> {{ $location->name }}</option>
                                    @endforeach
                                </select>
                            </div>
                            <x-jet-input-error for="rows.{{$index}}.location" class="text-theme-6 text-xs"/>
                        </td>
                        <td class="whitespace-nowrap">
                            <button wire:click="removeRow({{$index}})" class="btn btn-primary">
                                X
                            </button>
                        </td>
                    </tr>
                @endforeach
                </tbody>
            </table>
            <div>
                <a wire:click="saveForm"
                   class="btn btn-primary">Save
                </a>
            </div>
        </div>

    </div>
</div>
`

**Comment by rashid1125** [comment]
Please can you share with me how to add new rows with data and also select data to get.?
I want to like this can you guide me about.?
